uhabits-android/src/main/java/org/isoron/uhabits/activities/common/dialogs/MultipleDialogsHandler.kt

```diff
@@ -0,0 +1,15 @@
+package org.isoron.uhabits.activities.common.dialogs
+
+import android.app.Dialog
+import java.lang.ref.WeakReference
+
+class MultipleDialogsHandler {
```

Comment:
This class is redundant. You can remove it, then move the companion object code to the top level:

```kotlin
import android.app.Dialog [...]

var currentDialog: ...

fun Dialog.dismissCurrent() { ...}
```
```